src/node.rs

```diff
@@ -87,255 +53,232 @@ impl Default for Taffy {
     }
 }
 
+#[allow(clippy::iter_cloned_collect)] // due to no-std support, we need to use `iter_cloned` instead of `collect`
 impl Taffy {
     /// Creates a new [`Taffy`]
     ///
     /// The default capacity of a [`Taffy`] is 16 nodes.
     #[must_use]
     pub fn new() -> Self {
-        Default::default()
+        Taffy::default()
     }
 
     /// Creates a new [`Taffy`] that can store `capacity` nodes before reallocation
+    #[must_use]
     pub fn with_capacity(capacity: usize) -> Self {
         Self {
-            id: INSTANCE_ALLOCATOR.allocate(),
-            allocator: Allocator::new(),
-            nodes_to_ids: new_map_with_capacity(capacity),
-            ids_to_nodes: new_map_with_capacity(capacity),
-            forest: Forest::with_capacity(capacity),
+            nodes: SlotMap::with_capacity(capacity),
+            children: SlotMap::with_capacity(capacity),
+            parents: SlotMap::with_capacity(capacity),
         }
     }
 
-    /// Allocates memory for a new node, and returns a matching generated [`Node`]
-    fn allocate_node(&mut self) -> Node {
-        let local = self.allocator.allocate();
-        Node { instance: self.id, local }
-    }
-
-    /// Stores a new node in the tree
-    fn add_node(&mut self, node: Node, id: NodeId) {
-        let _ = self.nodes_to_ids.insert(node, id);
-        let _ = self.ids_to_nodes.insert(id, node);
-    }
-
-    /// Returns the `NodeId` of the provided node within the forest
-    fn find_node(&self, node: Node) -> Result<NodeId, error::InvalidNode> {
-        match self.nodes_to_ids.get(&node) {
-            Some(id) => Ok(*id),
-            None => Err(error::InvalidNode(node)),
-        }
-    }
+    /// Creates and adds a new unattached leaf node to the tree, and returns the [`NodeId`] of the new node
+    pub fn new_leaf(&mut self, layout: FlexboxLayout) -> TaffyResult<Node> {
+        let id = self.nodes.insert(NodeData::new(layout));
+        let _ = self.children.insert(new_vec_with_capacity(0));
```

Comment:
What is the reason for the dropped binding here?

Reply:
The key for each slotmap ends up the same but .insert is labeled must_use so either we do a drop binding or we call drop manually.

I turned on clippy pedantic just to poke around and it suggested an explicit drop... in any case all of the keys are the same.